### DISCUSSION TITLE
Outlook does not send email

### DISCUSSION THREAD

**Discussion by sucremad** [cmu-sei/GHOSTS#293] [Q&A] [answered]
Hi,

I am trying to use sample outlook timeline with small changes, it prepares a draft but does not send the mail. I set up a local AD lab environment, the user ghost uses is administrator. Tried after turn off the defender and firewall also. Here is the timeline. What am I doing wrong? Could you help please?

```json
{
    "Status": "Run",
    "TimeLineHandlers": [
      {
        "HandlerType": "Outlook",
        "Initial": "",
        "UtcTimeOn": "00:00:00",
        "UtcTimeOff": "24:00:00",
        "Loop": "True",
        "TimeLineEvents": [
          {
            "Command": "create",
            "CommandArgs": [
              "CurrentUser",
              "user@domain.com",
              "user3@domain.com",
              "",
              "test subject",
              "Ghost test",
              "PlainText",
              ""
            ],
            "DelayAfter": 900000,
            "DelayBefore": 0
          },
          {
            "Command": "clickrandomlink",
            "CommandArgs": [],
            "DelayAfter": 900000,
            "DelayBefore": 0
          }
        ]
      },
      {
        "HandlerType": "Outlook",
        "Initial": "",
        "UtcTimeOn": "00:00:00",
        "UtcTimeOff": "24:00:00",
        "Loop": "True",
        "TimeLineEvents": [
          {
            "Command": "create",
            "CommandArgs": [
              "CurrentUser",
              "random",
              "random",
              "random",
              "Random",
              "Random",
              "PlainText",
              ""
            ],
            "DelayAfter": 900000,
            "DelayBefore": 0
          },
          {
            "Command": "clickrandomlink",
            "CommandArgs": [],
            "DelayAfter": 900000,
            "DelayBefore": 0
          }
        ]
      }
    ]
  }
```

<img width="1165" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/44017937/4b078f01-a104-4fec-b228-32264ea39e72">

**Comment by sei-dupdyke** [comment]
Hi, thanks for checking out the project.

What version of Outlook are you using? I suspect it is a newer version. There is a known issue with newer versions and the version of Redemption that we are using to manage certain Outlook functions. I am in the process of getting the library updated, this should be very soon hopefully.

**Reply by sucremad** [reply]
Hi. I am using Outlook 2013.

**Reply by sucremad** [reply]
Do I have to install an API server?

**Reply by sei-dupdyke** [reply]
No, the API is not required. Still trying to get that library updated, sorry for the delay.

**Reply by sucremad** [reply]
Thank you, no problem. I have one more question.

In the logs it says:

`Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|System.Runtime.InteropServices.COMException: Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))`

I am trying on Windows Server 2022 Datacenter 21H2. This server or the version not supported?

**Reply by sei-dupdyke** [reply]
Ah — that might be your issue in this particular case. I have not tested with that version. Typically people are using the basic Win10 "user" versions. Sorry.

**Reply by sucremad** [reply]
this is the whole part:

2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Redemption64 loaded from C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption64.dll
2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Redemption loaded from C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption.dll
2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Creating new RDO session...
2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Redemption64 loaded from C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption64.dll
2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Redemption loaded from C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption.dll
2024/03/27 15:53:29.392|Ghosts.Client.Handlers.Outlook..ctor|Creating new RDO session...
2024/03/27 15:53:29.407|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Redemption is using C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption64.dll
2024/03/27 15:53:29.802|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Attempting register of Redemption COM
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Loading redemption failed 32 - C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption.dll
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Loading redemption failed 64 - C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption64.dll
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|System.Runtime.InteropServices.COMException: Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Loading redemption failed 32 - C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption.dll
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|Loading redemption failed 64 - C:\Users\Public\Downloads\ghosts-client-x64-v8\ghosts-client-x64-v8.0.0\lib\redemption64.dll
2024/03/27 15:53:30.226|Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject|System.Runtime.InteropServices.COMException: Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
2024/03/27 15:53:30.226|Ghosts.Client.Handlers.Outlook..ctor|RDO load error: System.Runtime.InteropServices.COMException (0x800401FA): Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.IClassFactory.CreateInstance(Object pUnkOuter, Guid& refiid, Object& ppunk)
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject(Guid guid) in C:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 256
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.new_RDOSession() in C:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 27
   at Ghosts.Client.Handlers.Outlook..ctor(TimelineHandler handler) in C:\Projects\ghosts\src\Ghosts.Client\Handlers\Outlook.cs:line 48
2024/03/27 15:53:30.226|Ghosts.Client.Handlers.Outlook..ctor|RDO load error: System.Runtime.InteropServices.COMException (0x800401FA): Wrong OS or OS version for application (Exception from HRESULT: 0x800401FA (CO_E_WRONGOSFORAPP))
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.IClassFactory.CreateInstance(Object pUnkOuter, Guid& refiid, Object& ppunk)
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.NewRedemptionObject(Guid guid) in C:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 256
   at Ghosts.Client.Infrastructure.Email.RedemptionLoader.new_RDOSession() in C:\Projects\ghosts\src\Ghosts.Client\Infrastructure\Email\RedemptionLoader.cs:line 27
   at Ghosts.Client.Handlers.Outlook..ctor(TimelineHandler handler) in C:\Projects\ghosts\src\Ghosts.Client\Handlers\Outlook.cs:line 48
2024/03/27 15:53:32.782|Ghosts.Client.Handlers.Outlook..ctor|Launching Outlook...
2024/03/27 15:53:33.616|Ghosts.Domain.Code.WorkingHours.Is|For Outlook: Local time: 15:53:33.6162988 UTC: 15:53:33.6162988 On: 00:00:00 Off: 24.00:00:00 Overnight? False
2024/03/27 15:53:33.616|Ghosts.Client.Handlers.Outlook.ExecuteEvents|DelayBefore sleeping for 900000 ms
2024/03/27 15:53:33.876|Ghosts.Client.Handlers.Outlook..ctor|Launching Outlook...
2024/03/27 15:53:34.095|Ghosts.Domain.Code.WorkingHours.Is|For Outlook: Local time: 15:53:34.0953084 UTC: 15:53:34.0953084 On: 00:00:00 Off: 24.00:00:00 Overnight? False
2024/03/27 15:53:34.095|Ghosts.Client.Handlers.Outlook.ExecuteEvents|DelayBefore sleeping for 900000 ms
2024/03/27 15:53:57.542|Quartz.Logging.LogProviders.NLogLogProvider+NLogLogger.Log|Batch acquisition of 0 triggers



Although I have the dlls in the lib folder it gives error:
<img width="667" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/44017937/45f2ea57-db55-485e-9ae7-b2afe3fbaacf">

**Comment by sei-dupdyke** [comment]
Did you register Redemption? https://www.systoolsfaqs.com/outlook/register-redemption-dll.html

**Reply by sucremad** [reply, marked as answer]
Finally solved the problem. I was using 64-bit ghosts with 32-bit office. Using 32-bit ghosts now works. That means it works on server 2022 too. Thank you so much.
